R/assignment_linter.R

```diff
@@ -88,7 +91,7 @@ assignment_linter <- function(allow_cascading_assign = TRUE,
 
   xpath <- paste(collapse = " | ", c(
     # always block = (NB: the parser differentiates EQ_ASSIGN, EQ_SUB, and EQ_FORMALS)
-    "//EQ_ASSIGN",
+    if (allow_equal_assign) "//LEFT_ASSIGN" else "//EQ_ASSIGN",
```

Comment:
Won't this also lint data.table code if `allow_equal_assign` is True? e.g. `dt[, x := 42L]`.

Reply:
That's a great catch that made me understand what `[text() = '...']` does. The latest commit should fix it. I included a test as well.